include/range/v3/view_interface.hpp

```diff
@@ -297,6 +297,19 @@ namespace ranges
             {
                 return Slice{}(detail::move(derived()), offs.from, offs.to);
             }
+            /// Returns a reference to the element at specified location pos, with bounds checking.
+            template<typename D = Derived,
+                CONCEPT_REQUIRES_(Same<D, Derived>() && RandomAccessRange<D>() && SizedRange<D>())>
+            auto at(range_difference_type_t<D> n) ->
+                decltype(std::declval<D &>().begin()[n])
+            {
+                using size_type = range_size_type_t<Derived>;
+                if (n < 0 || size_type(n) >= ranges::size(derived()))
+                {
+                    throw std::out_of_range("view_interface::at");
+                }
+                return derived().begin()[n];
+            }
```

Comment:
> This needs a const overload

Well that's embarrassing. I apologize for the oversight. Correction incoming.

> maybe make them lvalue ref qualified

C++ 14 has spoiled me. Is it not the case that the trailing return type on line 304 would respect const-ness, ref-ness, and violatile-ness? Or have I misunderstood your feedback?

```cpp
decltype(std::declval<D &>().begin()[n])
```